internal/datastore/crdb/reader.go

```diff
@@ -143,6 +143,23 @@ func (cr *crdbReader) LookupNamespaces(ctx context.Context, nsNames []string) ([
 	return nsDefs, nil
 }
 
+func (cr *crdbReader) QueryRelationshipsForDirectCheck(
+	ctx context.Context,
+	filter datastore.DirectCheckRelationshipsFilter,
+	opts ...options.QueryOptionsOption,
+) (iter datastore.RelationshipIterator, err error) {
+	qBuilder := common.NewSchemaQueryFilterer(schema, queryTuples).FilterWithDirectCheckFilter(filter)
```

Comment:
Could we find a way to unify the filter interface, so that we don't need both `QueryRelationshipsForDirectCheck` and `QueryRelationships`?

Reply:
instead of structs, maybe filters are `func(querybuilder)` so they can add their own clauses

Reply:
Can't have the caller adding their own clauses; not every datastore is SQL, unfortunately